### PR TITLE
Create a SwiftUI demo app, and consume Fluent via Swift Package

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [ ] iOS
- [ ] visionOS
- [ ] macOS

### Description of changes

This creates a brand new cross-platform demo app in SwiftUI, with support for iOS and macOS. This demo consumes Fluent via its `Package.swift` file, as opposed to the existing demo apps that consume static libraries.

No real demo controllers yet--just verifying that the project can be built, linked, and run via SPM.

iOS currently features a SwiftUI `Button` styled via `FluentButtonStyle`. There is no SwiftUI in FluentUI-macOS yet, so we are instead hosting an existing `NSButton` subclass, `FluentUI.Button`, inside an `NSViewRepresentable`.

### Binary change

None--no changes were made to the core libraries.

### Verification

Launched both iOS and macOS variants of the demo app.

<details>
<summary>Visual Verification</summary>

| macOS                                       | iOS                                      |
|----------------------------------------------|--------------------------------------------|
| macOS | iOS |

</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [x] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/2075)

### CODE REVIEW

File: Demo/FluentUIDemo/ContentView.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+import FluentUI
+
+struct ContentView: View {
+#if os(macOS)
+    // Until we have a SwiftUI button on macOS, this will have to do.
+    private struct ButtonRepresentable: NSViewRepresentable {
```
Comment (joannaquu) [inline Demo/FluentUIDemo/ContentView.swift:12]: no macOS swiftui button but we could use notification if we want to rip out the view representable

**Review by joannaquu — APPROVED** [review]

**Review by nodes11 — APPROVED** [review]

File: Demo/FluentUIDemo/ContentView.swift (hunk 1 above)
Comment (mischreiber) [reply Demo/FluentUIDemo/ContentView.swift:12]: Yeah I totally forgot about that component! If you're okay with it, maybe I'll save that for the first real demo page once we add a proper navigation flow to the app?

File: Demo/FluentUIDemo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
+				LOCALIZATION_PREFERS_STRING_CATALOGS = YES;
+				MTL_ENABLE_DEBUG_INFO = INCLUDE_SOURCE;
+				MTL_FAST_MATH = YES;
+				ONLY_ACTIVE_ARCH = YES;
+				SWIFT_ACTIVE_COMPILATION_CONDITIONS = "DEBUG $(inherited)";
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+			};
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply Demo/FluentUIDemo.xcodeproj/project.pbxproj:234]: Agreed--will be refactoring existing xcconfigs for the current iOS demo app for reuse here.

File: Demo/FluentUIDemo/ContentView.swift [hunk 3]
```diff
@@ -0,0 +1,42 @@
+//
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply Demo/FluentUIDemo/ContentView.swift:1]: I actually do intend for these to share a single root content view: a `NavigationSplitView` whose data will be piped in via platform-specific demo lists.

**Pull request merged**